diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index 8288fd234..b52fcfa9a 100755
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -85,10 +85,20 @@ elseif(ARM64MATCH GREATER "-1")
     message(STATUS "Detected ARM64 target processor")
     set(ARM64 1)
 
-    # Options for cross compiling AArch64 optional extensions
-    option(CROSS_COMPILE_SVE "Cross Compile for SVE Target" OFF)
-    option(CROSS_COMPILE_SVE2 "Cross Compile for SVE2 Target" OFF)
-    option(CROSS_COMPILE_NEON_DOTPROD "Cross Compile for Neon DotProd Target" OFF)
+    list(FIND ARM64_ALIASES "${CMAKE_HOST_SYSTEM_PROCESSOR}" NATIVE_ARCH)
+    if(NATIVE_ARCH EQUAL "-1")
+        message(STATUS "Cross compiling for ARM64 arch")
+        option(CROSS_COMPILE_SVE "Cross Compile for SVE Target" OFF)
+        option(CROSS_COMPILE_SVE2 "Cross Compile for SVE2 Target" OFF)
+        option(CROSS_COMPILE_NEON_DOTPROD "Cross Compile for Neon DotProd Target" ON)
+        set(CROSS_COMPILE_ARM64 1)
+    else()
+        # Options for cross compiling AArch64 optional extensions
+        option(CROSS_COMPILE_SVE "Cross Compile for SVE Target" OFF)
+        option(CROSS_COMPILE_SVE2 "Cross Compile for SVE2 Target" OFF)
+        option(CROSS_COMPILE_NEON_DOTPROD "Cross Compile for Neon DotProd Target" OFF)
+        set(CROSS_COMPILE_ARM64 0)
+    endif()
 else()
     message(STATUS "CMAKE_SYSTEM_PROCESSOR value `${CMAKE_SYSTEM_PROCESSOR}` is unknown")
     message(STATUS "Please add this value near ${CMAKE_CURRENT_LIST_FILE}:${CMAKE_CURRENT_LIST_LINE}")
@@ -305,7 +315,13 @@ if(GCC)
             set(ARM_ARGS -O3 -march=armv9-a+sve2)
             add_definitions(-DHAVE_SVE2=1)
         endif()
-        set(ARM_ARGS ${ARM_ARGS} -fPIC -flax-vector-conversions)
+        set(ARM_ARGS ${ARM_ARGS} -flax-vector-conversions)
+        if(NOT MINGW)
+            set(ARM_ARGS ${ARM_ARGS} -fPIC)
+        endif()
+        if(APPLE)
+            set(ARM_ARGS ${ARM_ARGS} -arch ${CMAKE_OSX_ARCHITECTURES})
+        endif()
         if(CPU_HAS_SVE)
             set(SVE_HEADER_TEST "
 #ifndef __ARM_NEON_SVE_BRIDGE
